xugrid/regrid/structured.py

```diff
@@ -21,56 +28,619 @@ class StructuredGrid1d:
+        there is no interpolation possible.
+
+        Args:
+            self (StructuredGrid1d): source grid
+            other (StructuredGrid1d): target grid
+
+        Returns:
+            valid_self_index (np.array): valid source indexes
+            valid_other_index (np.array): valid target indexes
+        """
+        source_index, target_index = self.valid_nodes_index(other)
+        valid = (other.midpoints[target_index] > self.midpoints[0]) & (
+            (other.midpoints[target_index] < self.midpoints[-1])
+        )
+        return source_index[valid], target_index[valid]
+
+    def overlap_1d_structured(self, other):
+        """
+        Returns source and target nodes and overlapping length. It utilises overlap_1d()
+        and does an aditional flip in cases of reversed midpoints
+
+        Args:
+            self (StructuredGrid1d): source grid
+            other (StructuredGrid1d): target grid
+
+        Returns:
+            valid_self_index (np.array): valid source indexes
+            valid_other_index (np.array): valid target indexes
+            weights (np.array): length of overlap
+        """
         source_index, target_index, weights = overlap_1d(self.bounds, other.bounds)
+        source_index = self.flip_if_needed(source_index)
+        target_index = other.flip_if_needed(target_index)
+        return source_index, target_index, weights
+
+    def centroids_to_linear_indexes(
+        self, other, source_index: np.array, target_index: np.array, weights: np.array
+    ):
+        """
+        Returns for every target node an pair of connected source nodes based on
+        centroids connection inputs
+
+        Args:
+            self (StructuredGrid1d): source grid
+            other (StructuredGrid1d): target grid
+            source_index (np.array): source index (centroids)
+            target_index (np.array): target index (centroids)
+            weights (np.array): weights (centroids)
+
+        Returns:biliniar
+
+            source_index (np.array): source index (linear)
+            target_index (np.array): target index (linear)
+            weights (np.array): weights (linear)
+        """
+        # in cases where midpoint(target) < midpoint(source), within bounding box of source,
+        # neighbor will be -1. All other cases +1
+        neighbour = np.ones(source_index.size,dtype=int)
+        neighbour[other.midpoints[target_index] < self.midpoints[source_index]] = -1
+        source_index = np.column_stack((source_index, source_index + neighbour)).ravel()
+        target_index = np.repeat(target_index, 2)
+        weights = np.column_stack((weights, 1.0 - weights)).ravel()
+        # correct for possibility of out of bound due to column-stack source_index + 1 and -1
+        valid = np.logical_and(source_index <= self.size - 1, source_index >= 0)
+        return source_index[valid], target_index[valid], weights[valid]
+
+    def compute_weights(self, other, source_index, target_index):
```

Comment:
the function name is not clear. Does it compute weights for inverse-to-distance interpolation? or for some other linear interpolation method?